src/network_models/power_flow_evaluation.jl

```diff
@@ -106,36 +123,99 @@ function _make_pf_input_map!(
         pf_data_opt_container_map = Dict{OptimizationContainerKey, map_type}()
         @info "Adding input map to send $category to $(nameof(typeof(pf_data)))"
         precedence = PF_INPUT_KEY_PRECEDENCES[category]
-        added_injection_types = DataType[]
-        # For each data source that is relevant to this category in order of precedence,
-        # loop over the component types where data exists at that source and record the
-        # association
-        for entry_type in precedence
-            for (key, val) in available_keys
-                if get_entry_type(key) === entry_type
-                    comp_type = get_component_type(key)
-                    # Skip types that have already been handled by something of higher precedence
-                    if comp_type in added_injection_types
-                        continue
-                    end
-                    push!(added_injection_types, comp_type)
-
-                    name_bus_ix_map = map_type()
-                    comp_names =
-                        if (key isa ParameterKey)
-                            get_component_names(get_attributes(val))
-                        else
-                            axes(val)[1]
-                        end
-                    for comp_name in comp_names
-                        name_bus_ix_map[comp_name] =
-                            temp_component_map[comp_type][comp_name]
+        _add_category_to_map!(
+            precedence,
+            available_keys,
+            temp_component_map,
+            pf_data_opt_container_map,
+        )
+        pf_e_data.input_key_map[category] = pf_data_opt_container_map
+    end
+    # the function to map HVDC power transfers as bus injections is not applicable in this case:
+    pf_data isa PFS.PSSEExporter && return
```

Comment:
use multiple dispatch and the type hierarchy for this (e.g., define trivial `_add_two_terminal_elements_map!` for `SystemPowerFlowContainer`)